rpc/rpc-wrappers.go

```diff
@@ -387,6 +388,37 @@ func NewValidationAuthorityServer(serverQueue string, channel *amqp.Channel, imp
 		return nil
 	})
 
+	rpc.Handle(MethodCheckCAARecords, func(req []byte) []byte {
+		var caaReq struct {
+			Ident core.AcmeIdentifier
+		}
+		if err := json.Unmarshal(req, &caaReq); err != nil {
+			// AUDIT[ Improper Messages ] 0786b6f2-91ca-4f48-9883-842a19084c64
+			improperMessage(MethodCheckCAARecords, err, req)
+			return nil
+		}
+
+		present, valid, err := impl.CheckCAARecords(caaReq.Ident)
+		if err != nil {
+			// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
+			errorCondition(MethodCheckCAARecords, err, caaReq)
+		}
+
+		var caaResp struct {
+			Present bool
+			Valid   bool
+			Err     error
+		}
+		caaResp.Present = present
+		caaResp.Valid = valid
+		caaResp.Err = err
+		response, err := json.Marshal(caaResp)
+		if err != nil {
```

Comment:
I know it's not in the current version, but the RPC update logs an errorCondition here if err != nil too